### PR TITLE
test: merge waitToDelete* functions

### PR DESCRIPTION
**Pull request by aanm**
Follow up of https://github.com/cilium/cilium/pull/10520#discussion_r391665734

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/10559)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

File: test/helpers/kubectl.go [hunk 1]
```diff
@@ -1519,9 +1519,12 @@ func (kub *Kubectl) waitToDeleteCiliumOperator() error {
 		default:
```
Comment (brb) [inline test/helpers/kubectl.go:1519]: At the line 1518 (GH does not allow me to comment it), you probably want to replace `"Cilium Operator"` with `name`.

File: test/helpers/kubectl.go [hunk 2]
```diff
@@ -1498,10 +1498,10 @@ func (kub *Kubectl) DeleteCiliumOperator() error {
 	// incomplete teardown.
 
 	_ = kub.DeleteResource("deployment", fmt.Sprintf("-n %s cilium-operator", GetCiliumNamespace(GetCurrentIntegration())))
-	return kub.waitToDeleteCiliumOperator()
+	return kub.waitToDelete("Cilium Operator", "io.cilium/app=cilium")
```
Comment (brb) [inline test/helpers/kubectl.go]: Can we have consts for `CiliumOperatorLabal` and `CiliumAgentLabel`?

**Review by brb — CHANGES_REQUESTED** [review]

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Review by brb — APPROVED** [review]

**Pull request merged**